src/Servers/HttpSys/src/RequestProcessing/RequestHeaders.Generated.tt

```diff
@@ -209,5 +209,36 @@ namespace Microsoft.AspNetCore.HttpSys.Internal
             }
 <# } #>
         }
+
+        private ReadOnlySpan<byte> HeaderKeys => new byte[]
+        {
+<# foreach(var prop in props) { #>
+            (byte)<#=prop.ID#>,
+<# } #>
+        };
+
+        private string GetHeaderKeyName(HttpSysRequestHeader header) => header switch
+        {
+<# foreach(var prop in props) { #>
+            <#=prop.ID#> => HeaderNames.<#=prop.Name#>,
+<# } #>
+            _ => header.ToString()
```

Comment:
I could not decide if throwing an exception or returning this string here is the better choice.

Reply:
ack, updating